modules/redis/redis.c

```diff
@@ -221,6 +212,40 @@ redis_dd_disconnect(LogThreadedDestDriver *s)
   self->c = NULL;
 }
 
+static void
+_fill_argv_from_template_list(RedisDriver *self, LogMessage *msg)
+{
+  ScratchBuffersMarker marker;
+  GString *buffer = scratch_buffers_alloc_and_mark(&marker);
+
+  for (gint i = 1; i < self->argc; i++)
+    {
+      log_template_format(g_list_nth_data(self->fields, i-1), msg, &self->template_options, LTZ_SEND,
+                          self->super.worker.instance.seq_num, NULL, buffer);
+      self->argvlen[i] = buffer->len;
+      self->argv[i] = g_strdup(buffer->str);
+    }
+  scratch_buffers_reclaim_marked(marker);
+}
+
+static void
+_free_argv_templated_part(RedisDriver *self)
+{
+  for (gint i = 1; i < self->argc; i++)
+    g_free(self->argv[i]);
+}
+
+static GString *
+_argv_to_string(RedisDriver *self, ScratchBuffersMarker *marker)
+{
+  GString *full_command = scratch_buffers_alloc_and_mark(marker);
+
+  full_command = g_string_append(full_command, self->argv[0]);
+  for (gint i = 1; i < self->argc; i++)
+    g_string_append_printf(full_command, " \"%s\"", self->argv[i]);
```

Comment:
Absolutely optional:

`argv[i]` is not escaped between `\"%s\"`.

You can use `lib/utf8utils` to escape them if you want to. An example from the JSON module:
```
static inline void
tf_json_append_escaped(GString *dest, const gchar *str, gssize str_len)
{
  append_unsafe_utf8_as_escaped_text(dest, str, str_len, "\"");
}
```

Reply:
Thanks, I fixed it! :)

Reply:
Nice, thanks! :)